airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java

```diff
@@ -171,7 +175,11 @@ protected JsonNode formatData(final FieldList fields, final JsonNode root) {
   protected void formatDateTimeFields(final FieldList fields, final JsonNode root) {
     final List<String> dateTimeFields = BigQueryUtils.getDateTimeFieldsFromSchema(fields);
     if (!dateTimeFields.isEmpty() && !root.isNull()) {
-      BigQueryUtils.transformJsonDateTimeToBigDataFormat(dateTimeFields, (ObjectNode) root);
+      if (root.isArray()) {
+        root.forEach(jsonNode -> BigQueryUtils.transformJsonDateTimeToBigDataFormat(dateTimeFields, (ObjectNode) jsonNode));
```

Comment:
does this work for double-nested arrays? E.g. if `root` looks like this:
```json
[
  [{"foo": "bar"}, {"foo": "baz"}]
]
```

Reply:
good point.
Fix is provided + few test cases:
![Screenshot from 2022-07-04 18-19-52](https://user-images.githubusercontent.com/30464745/177182961-fda70db6-f714-4f02-bedb-74cf2fb31ae4.png)